crmsh/corosync.py

```diff
@@ -58,17 +59,405 @@ def __repr__(self):
 
 
 class QDevice(object):
-    def __init__(self, ip, port=5403, algo="ffsplit", tie_breaker="lowest"):
+    '''
+    Whole certification process:
+    For init
+    Step 1:  init_db_on_qnetd
+    Step 2:  fetch_qnetd_crt_from_qnetd
+    Step 3:  copy_qnetd_crt_to_cluster
+    Step 4:  init_db_on_cluster
+    Step 5:  create_ca_request
+    Step 6:  copy_crq_to_qnetd
+    Step 7:  sign_crq_on_qnetd
+    Step 8:  fetch_cluster_crt_from_qnetd
+    Step 9:  import_cluster_crt
+    Step 10: copy_p12_to_cluster
+    Step 11: import_p12_on_cluster
+
+    For join
+    Step 1:  fetch_qnetd_crt_from_cluster
+    Step 2:  init_db_on_local
+    Step 3:  fetch_p12_from_cluster
+    Step 4:  import_p12_on_local
+    '''
+    def __init__(self, ip, port=5403, algo="ffsplit",
+                 tie_breaker="lowest", tls="on", cluster_node=None, cmds=None):
         self.ip = ip
         self.port = port
         self.algo = algo
         self.tie_breaker = tie_breaker
+        self.tls = tls
+        self.cluster_node = cluster_node
+        self.askpass = False
+        self.cmds = cmds
+
+        self.qnetd_service = "corosync-qnetd.service"
```

Comment:
I think we could put all of them as constants outside the `__init__`. They don't like that can be changed, right?

Reply:
OK
changed